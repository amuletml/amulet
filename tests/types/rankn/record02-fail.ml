external val (+) : int -> int -> int = ""

type foo = Foo of { bar : forall 'a. 'a -> 'a }

let foo = Foo { bar = fun x -> x + 1 }

let main (Foo r) = r.bar 1
