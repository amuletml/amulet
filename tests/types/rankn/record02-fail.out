record02-fail.ml[5:17 ..5:36]: error
  Could not match the rigid type variable 'a with the type int

  • When checking that the type
     int -> int
    can be made as polymorphic as
     forall 'a. 'a -> 'a

  • Arising in the this field bar
  │ 
5 │ let foo = Foo { bar = fun x -> x + 1 }
  │                 ^^^^^^^^^^^^^^^^^^^^
